Added Amazon Linux Support. Bumped chef and supported OS versions

rwinfosec opened the pull request:
*Bumped to Chef 14.
*Added Support for Amazon Linux 2018.03 and 2
*Updated and fixed support for Ubuntu 16.04 and 18.04
*Updated support for OSX to 10.13

jacknagz commented on recipes/amazon.rb:
```diff
@@ -0,0 +1,36 @@
+# Cookbook Name:: osquery
+# Recipe:: amazon
+#
+# Copyright 2019, Dwolla Inc.
```
Can you remove this copyright? It goes against the Apache 2 License

rwinfosec replied:
Yep! No problem.

jacknagz commented on LICENSE:
```diff
@@ -0,0 +1,14 @@
+Copyright 2013-2014 Jack Naglieri <jacknagzdev@gmail.com>
```
I'll add the LICENSE in a follow-up PR, along with CONTRIBUTING, and more.

rwinfosec replied:
With the LICENSE file, it is required by FoodCritic and will fail the TravisCI build. I will just add an empty LICENSE file.

rwinfosec commented:
Regards to testing, I've updated some of the unit tests to reflect the versions that I am testing.
I've verified all integration tests are passing with each vagrant image that is supported and added an additional integration test to verify that the osquery configuration file JSON content is as expected. I've included the changes to this cookbook into another and converged with an AWS Linux 2018.03 instance in AWS to verify that it is working as intended.
